server/src/main/java/org/opensearch/search/DefaultSearchContext.java

```diff
@@ -909,16 +920,39 @@ public boolean shouldUseConcurrentSearch() {
      * Evaluate if parsed request supports concurrent segment search
      */
     public void evaluateRequestShouldUseConcurrentSearch() {
-        if (sort != null && sort.isSortOnTimeSeriesField()) {
+
+        if (getConcurrentSearchMode().equals(CONCURRENT_SEGMENT_SEARCH_MODE_ALL)) {
+            // if the mode is set to all, enable concurrent search for all requests
+            requestShouldUseConcurrentSearch.set(canEnableConcurrentSearch());
+            return;
+        } else if (getConcurrentSearchMode().equals(CONCURRENT_SEGMENT_SEARCH_MODE_NONE)) {
+            // mode is set to none, disable concurrent search for all requests
             requestShouldUseConcurrentSearch.set(false);
+            return;
+        }
+
+        // when mode is set to "auto" evaluate from the deciders by visiting the query
+        ConcurrentSearchDeciders concurrentSearchDeciders = new ConcurrentSearchDeciders(
+            this.concurrentSearchDeciders,
+            this,
+            clusterService.getClusterSettings(),
+            indexService.getIndexSettings()
+        );
+        requestShouldUseConcurrentSearch.set(concurrentSearchDeciders.getConcurrentSearchDecisionForSearchRequest());
+    }
+
+    @Override
+    public boolean canEnableConcurrentSearch() {
```

Comment:
```suggestion
    public boolean parsedRequestSupportsConcurrentSearch() {
```

I think we want to indicate that the factors used here are parsed from the search request.